### ISSUE TITLE
Getting Illegal character in path when we use %20 or space  in url path

### ISSUE THREAD

**Issue by AnkitKadam567** [specmatic/specmatic#905]
**Description**
We need to pass path parameter with spaces in url and to achieve this we are replacing %20 with empty space. But this is giving us Illegal character in path. Is there any other way to pass empty space in URL?

**Steps to reproduce**
1. Use url which contains space in path parameter
2. run test

**Expected behavior**
URL with empty space should have executed without any issue / error

**Screenshots**
Error Message
 ```
 Illegal character in path at index 84: http://localhost:3000/service/platform/insights/v1.0/company/1/template/validate/RTO Report
  " Scenario: GET /service/platform/insights/v1.0/company/(company_id:number)/template/name/(template_name:string)/layout -> 200 | success FAILED"
        Reason: Testing scenario "Fetch layout from template name. Response: Response received when layout is fetched from template name"
    	API: GET /service/platform/insights/v1.0/company/(company_id:number)/template/name/(template_name:string)/layout -> 200
```

Use in .yml

   ```
 defaultTemplateName:
      name: default_template_name
      in: path
      required: true
      schema:
        type: string
      examples: 
        success:
          value: Returns%20Report
```

**System Information:**
 - OS : MacOS

**Comment by AnkitKadam567** [comment]
@harikrishnan83 @joelrosario Thank you both for your prompt responses. Could you please specify when and the  version in which this fix is expected to be available?

**Comment by harikrishnan83** [comment]
hello @AnkitKadam567, this fix is available as part of our latest [release 1.2.5](https://github.com/znsio/specmatic/releases/tag/1.2.5). cc @joelrosario
